akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -42,6 +42,7 @@ sealed trait HttpMessage extends jm.HttpMessage {
   def isResponse: Boolean
 
   def headers: immutable.Seq[HttpHeader]
+  def attributes: Map[AttributeKey[Any], Any]
```

Comment:
Why not `AttributeKey[_]` (not that we could properly model that in a map but it would at least be a bit more accurate)?

Reply:
I.e. we could probably remove part of the `instanceOf`s below.